packages/connect/src/connect-error.ts

```diff
@@ -130,6 +130,25 @@ export class ConnectError extends Error {
     return new ConnectError(String(reason), code, undefined, undefined, reason);
   }
 
+  static [Symbol.hasInstance](v: unknown): boolean {
+    if (Object.getPrototypeOf(v) === ConnectError.prototype) {
+      return true;
+    }
+    return (
+      v instanceof Error &&
+      v.name === "ConnectError" &&
+      "code" in v &&
+      Object.values(Code).includes(v.code as keyof Code) &&
+      "metadata" in v &&
+      v.metadata instanceof Headers &&
+      "details" in v &&
+      Array.isArray(v.details) &&
+      "rawMessage" in v &&
+      typeof v.rawMessage == "string" &&
+      "cause" in v
```

Comment:
`instanceof Headers` can be problematic because of polyfills. Let's remove this check.

Because TypeScript enums are lookup objects with numeric _and_ string values, let's just use `typeof v.code == "number"`.